awx/main/fields.py

```diff
@@ -652,6 +652,9 @@ def validate(self, value, model_instance):
                 )
                 if match:
                     label, extraneous = match.groups()
+                    # bail if the dependency is for a dynamic source
```

Comment:
For new credentials, these dynamic input relationships don't exist yet:

![Screenshot from 2019-09-21 09-59-20](https://user-images.githubusercontent.com/9753817/65374313-982e9a80-dc56-11e9-9067-99dd8d892407.png)

To get the expected validation behavior on for this, we'll need to come up with a way of knowing the intended dynamic inputs before they've been created.

For regular / static fields that are encrypted, we've dealt with _similar_ issues in the past by (ab)using the special `$encrypted` value. We may need to do something similar here.

Reply:
@omgjlk Once the prompted fields are saved, I suspect that an updated task container might be all that's needed to resolve https://github.com/ansible/awx/pull/4798#issuecomment-533712837 but it would be good to confirm - thanks for taking a look.